From 31bd7a0da3d0855a444bfcfc718a433c02ba1111 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Thu, 9 May 2019 03:51:42 +0200
Subject: [PATCH] Ensure the correct git repository is found by genversion.py

---
 genversion.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/genversion.py b/genversion.py
index d4c0919..1677b1c 100755
--- a/genversion.py
+++ b/genversion.py
@@ -22,6 +22,8 @@
 
 import os, subprocess, sys, argparse
 
+path_to_genversion = os.path.abspath(os.path.dirname(__file__))
+
 def removePrefix(s, prefix):
     if s.startswith(prefix):
         return s[len(prefix):]
@@ -219,6 +221,11 @@ def main():
         print("Cannot regenerate {0} from git".format(args.out))
         sys.exit(0)
 
+    # Ensure the release tarball isn't being built from inside a different git repository
+    if path_to_genversion != root_dir:
+        print("Cannot regenerate {0} from git".format(args.out))
+        sys.exit(0)
+
     commit_hash = sh("git rev-parse HEAD").strip()
 
     gitDescribe = GitDescribe(sh("git describe --dirty").strip())
-- 
2.21.0

